### DISCUSSION TITLE
vercel dev not serving dynamic catch-all routes as expected

### DISCUSSION THREAD

**Discussion by baked-dev** [vercel/vercel#7435] [Help]
hi, 
im working on a small function i want to deploy to my personal account and ran into the issue that dynamic catch all routes do not work as expected in development. I was wondering if I was doing something wrong.

Using Vercel CLI 23.1.3-canary.75

/api/[...param].ts

```ts
import type { VercelRequest, VercelResponse } from "@vercel/node";

export default (request: VercelRequest, response: VercelResponse) => {
  const { param } = request.query;
  console.log(request.query);
  response.status(200).send(`Hello ${param}!`);
};
```

when starting the dev server with "vercel dev" and requesting /api/test i expect my browser to show "Hello test!" and the console to log { 'param': ['test'} }, instead it shows "Hello undefined!" and logs { '...param': 'test' }.
when requesting /api/test/test2 the server returns a 404 page.

In a next.js project the routing works as expected and requests to /api/test/test2 correctly show "Hello test,test2!"

The documentation on vercel functions mentions nextjs api routes right in the beginning which to me indicated that the routing is supposed to work similar to it.

**Comment by LoicMahieu** [comment]
Same here...
We wanted to port some API routes from next to a standalone "vercel" project and it does not work...

**Comment by clarkeg02** [comment]
Same issue here. Everything works while hosting locally but dynamic and catch-all routes both fail after being deployed to vercel.
